src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -499,6 +499,54 @@ internal void Insert(ref T[] list, string? name, MemberListType listType)
                     }
                 }
 
+                private void MergeWithGlobalListInOrder(T[] list)
```

Comment:
Only called when `MemberListType.All`, other cases are only requesting one member or a members with same where it comes in order. The existing method  `MergeWithGlobalList(T[] list)` looks more optimal for those cases